src/main/java/com/yelp/nrtsearch/server/luceneserver/field/PolygofieldDef.java

```diff
@@ -0,0 +1,127 @@
+
+    if (requestField.getHighlight()) {
+      throw new IllegalArgumentException(
+          String.format(
+              "field: %s cannot have highlight=true. only type=text or type=atom fields can have highlight=true",
+              getName()));
+    }
+
+    if (requestField.getStore()) {
+      throw new IllegalArgumentException("polygon fields cannot be stored");
+    }
+
+    if (hasAnalyzer(requestField)) {
+      throw new IllegalArgumentException("no analyzer allowed on polygon field");
+    }
+  }
+
+  protected DocValuesType parseDocValuesType(Field requestField) {
+    // ES does not support geoshape doc value
+    // polygon can only be retrieved from _source in ES
+    // TODO: figure out later for nrtsearch
+    return DocValuesType.NONE;
+  }
+
+  @Override
+  public void parseDocumentField(
+      Document document, List<String> fieldValues, List<List<String>> facetHierarchyPaths) {
+    if (fieldValues.size() > 1) {
+      throw new IllegalArgumentException("polygon length cannot be more than 1.");
+    }
+    for (String fieldValue : fieldValues) {
+      /**
+       * Map<String, Object> geojson = gson.fromJson(fieldValue, Map.class); if
+       * (!geojson.get("type").equals("polygon")) { throw new IllegalArgumentException( "Non
+       * supported types. Only polygon type is supported." ); } List<List<List<Double>>> coordinates
+       * = (List<List<List<Double>>>) geojson.get("coordinates"); List<List<Double>> outer =
+       * coordinates.get(0); List<List<List<Double>>> holes = coordinates.subList(1,
+       * coordinates.size()); Polygon[] holesArray = (Polygon[]) ( holes.stream().map(hole ->
+       * generatePolygon(hole, new Polygon[0])) ).toArray();
+       */
+      Polygon[] polygons;
+      try {
+        polygons = Polygon.fromGeoJSON(fieldValue);
+      } catch (ParseException e) {
+        throw new RuntimeException("Invalid geojson " + fieldValue + "\nException: " + e);
+      }
+      if (polygons.length > 1) {
+        throw new IllegalArgumentException("Multipolygon not supported");
+      }
+
+      Arrays.stream(LatLonShape.createIndexableFields(getName(), polygons[0]))
+          .forEach(x -> document.add(x));
+    }
+  }
+
+  /**
+   * private Polygon generatePolygon(List<List<Double>> points, Polygon[] holes) { double[] lats =
+   * new double[points.size()]; double[] lngs = new double[points.size()]; for (int i = 0; i <
+   * points.size(); i++) { lngs[i] = points.get(i).get(0); lats[i] = points.get(i).get(1); } return
+   * new Polygon(lats, lngs, holes); }
+   */
+  @Override
+  public String getType() {
+    return "POLYGON";
+  }
+
+  @Override
+  public LoadedDocValues<?> getDocValues(LeafReaderContext context) throws IOException {
+    // TODO: figure out later
+    return null;
+  }
+
+  @Override
+  public Query getGeoPolygoQuery(GeoPolygonQuery geoPolygonQuery) {
+    return LatLonShape.newBoxQuery(
+        geoPolygonQuery.getField(),
+        ShapeField.QueryRelation.CONTAINS,
```

Comment:
Should we let users specify the `QueryRelation` for this query ?

Reply:
We only have polygon contains point use case. I think it is enough to make it simple like this